packages/astro/components/ViewTransitions.astro

```diff
@@ -75,6 +84,36 @@ const { fallback = 'animate' } = Astro.props as Props;
 			});
 		});
 
+		if(document.querySelector('[name="astro-view-transitions-forms"]')) {
+			document.addEventListener('submit', (ev) => {
+				let el = ev.target as HTMLElement;
+				if (
+				el.tagName !== 'FORM' ||
+					isReloadEl(el)
+				) {
+					return;
+				}
+
+				const form = el as HTMLFormElement;
+				const formData = new FormData(form);
+				const params = new URLSearchParams(formData as any); // TS is wrong here
+				let action = form.action;
+				const init: RequestInit = {
+					method: form.method,
+					redirect: 'follow'
+				} satisfies RequestInit;
+				if(form.method === 'get') {
+					const url = new URL(action);
+					url.search = params.toString();
+					action = url.toString();
+				} else {
+					init.body = params;
+				}
+				ev.preventDefault();
+				navigate(action, { init });
```

Comment:
We only need this for method === "post".
The navigate event of the navigation API has a formData property. 

The HTML Standard says:
> event.[formData](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigateevent-formdata)
>
>    The [FormData](https://xhr.spec.whatwg.org/#formdata) representing the submitted form entries for this navigation, if this navigation is a "[push](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigationtype-push)" or "[replace](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigationtype-replace)" navigation representing a POST [form submission](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-form-submit); null otherwise.
>
>    (Notably, this will be null even for "[reload](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigationtype-reload)" or "[traverse](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigationtype-traverse)" navigations that are revisiting a [session history entry](https://html.spec.whatwg.org/multipage/browsing-the-web.html#session-history-entry) that was originally created from a form submission.)

It would be great if we could do this similarly here, passing the form data as `options.formData` instead of the RequestInit object (only for "post" and nothing for "get")

Reply:
Yeah I agree. The one downside I see here is that formData has a Content-Type of `multipart/form-data` whereas the default of a form is `application/x-www-form-urlencoded` So we would be changing the default. Probably ok since Astro can handle both.

What happens though if the form has `enctype` set? Are we going to ignore that and just always use the formdata type? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#enctype